### PR TITLE
#368 : Hide/Show button for password input

### PR DESCRIPTION
**Pull request by ekkaaditya200**
## Description
Added a password visibility toggle (eye icon) for both the login and registration forms in the Electron Start Menu.

## Checklist

- [✅] I have read and followed the [Contributing Guidelines](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/docs/CONTRIBUTING.md).
- [✅] I have followed the project's [git convention](https://github.com/AntonioMrtz/SpotifyElectron/blob/master/docs/developer/Git-Convention.md) for the Pull Request's name and git commits.
- [✅] I have tested the changes and verified they work as intended. Automated tests (unit, integration, e2e as applicable) and manual tests (UI interactions, edge cases, error states, etc.).
- [✅] I have set the pull request base to the appropriate `development-vx.xx` branch corresponding to the issue milestone.

## Commit type
feat – This PR adds a new feature for toggling password visibility.

## Issue - 
Number - #368 

## Proposed Changes

Added a password visibility toggle eye icon in:
Electron/src/pages/StartMenu/StartMenu.tsx
Electron/src/pages/StartMenu/RegisterMenu.tsx

Added corresponding CSS styles in:
Electron/src/pages/StartMenu/startMenu.module.css

Added translation keys for the button in localization files:
Electron/src/i18n/localization/en/start-menu.json
Electron/src/i18n/localization/sn/start-menu.json

## Screenshots
Screenshot 2025-07-22 at 1 14 19 AM

Screenshot 2025-07-22 at 1 15 02 AM

## Assigned

@AntonioMrtz

### CODE REVIEW

**Comment by AntonioMrtz** [comment]
Hi @ekkaaditya200 :)
Do you think we can have the show/hide icon for both password and confirm password on register menu?

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
Tests for new functionality + hide/show for both password and confirm password on register screen

**Comment by ekkaaditya200** [comment]
Hi @AntonioMrtz 
I realized I mistakenly marked the testing checkbox in the PR checklist, I have not worked on testing before, so I apologize for the confusion. If needed you can assign the issue to someone else.
I would really appreciate it if you could suggest any resources or guidance to help me learn how to write tests. I am eager to improve and contribute properly.

**Comment by AntonioMrtz** [comment]
Hi @ekkaaditya200 . You can check the [official Jest Guide](https://jestjs.io/docs/tutorial-react). Previous testing knowledge is not needed for contributing. You can learn on the go for this task if you want. Let me know your decision 🫡

**Comment by ekkaaditya200** [comment]
I have completed it and will raise a pull request shortly.

**Comment by ekkaaditya200** [comment]
Hi @AntonioMrtz  I added the testing functions. You can review it.

File: Electron/src/pages/StartMenu/RegisterMenu.tsx [hunk 1]
```diff
@@ -199,33 +205,83 @@ export default function RegisterMenu({ setIsSigningUp }: PropsRegisterMenu) {
...
-                id="confirmpassword"
-                placeholder={t('registerMenu.form-password-repeat')}
-                onChange={handleChange}
-                disabled={loading}
-                spellCheck={false}
-                required
-              />
+              <div className={styles.passwordInputWrapper}>
+                <input
+                  type={confirmPasswordVisibility ? 'text' : 'password'} // Toggle based on state
```
Comment (AntonioMrtz) [inline Electron/src/pages/StartMenu/RegisterMenu.tsx:251]: Same as the other comment

**Review by AntonioMrtz — CHANGES_REQUESTED** [review]
Minor changes. Looks good to me :)

File: Electron/src/pages/StartMenu/RegisterMenu.tsx [hunk 2]
```diff
@@ -199,33 +205,83 @@ export default function RegisterMenu({ setIsSigningUp }: PropsRegisterMenu) {
...
-                name="password"
-                id="password"
-                placeholder={t('registerMenu.form-password')}
-                onChange={handleChange}
-                disabled={loading}
-                spellCheck={false}
-                required
-              />
+              <div className={styles.passwordInputWrapper}>
+                <input
```
Comment (AntonioMrtz) [inline Electron/src/pages/StartMenu/RegisterMenu.tsx:209]: Check if the same item align problem occurs here

File: Electron/src/pages/StartMenu/RegisterMenu.tsx [hunk 3]
```diff
@@ -199,33 +205,83 @@ export default function RegisterMenu({ setIsSigningUp }: PropsRegisterMenu) {
...
-                name="confirmpassword"
-                id="confirmpassword"
-                placeholder={t('registerMenu.form-password-repeat')}
-                onChange={handleChange}
-                disabled={loading}
-                spellCheck={false}
-                required
-              />
+              <div className={styles.passwordInputWrapper}>
+                <input
```
Comment (AntonioMrtz) [inline Electron/src/pages/StartMenu/RegisterMenu.tsx:250]: Check if the same item align problem occurs here
